src/messaging/ReliableMessageProtocolConfig.cpp

```diff
@@ -88,4 +88,14 @@ Optional<ReliableMessageProtocolConfig> GetLocalMRPConfig()
                                              : Optional<ReliableMessageProtocolConfig>::Value(config);
 }
 
+System::Clock::Timestamp GetRetransmissionTimeout(System::Clock::Timestamp baseInterval)
+{
+    System::Clock::Timestamp timeout(0);
+    for (uint8_t i = 0; i < CHIP_CONFIG_RMP_DEFAULT_MAX_RETRANS + 1; i++)
+    {
+        timeout += Messaging::ReliableMessageMgr::GetBackoff(baseInterval, i, /* useRandom */ false);
```

Comment:
This seems to assume that the base interval doesn't change between retransmissions while in theory, it might happen that the first retransmission uses the active interval, while subsequent ones happen long enough since the last RX from the peer node to use the idle interval. Not sure if that's a problem in practice (if used response timeouts are short enough for it to make a difference), but wanted to point it out. Maybe this function could take both base intervals and last peer actvity time to make sure the outcome is always an upper bound for the sum of retry intervals. Or we could assume idle at first and restart the response timer upon receiving an ack for the request...

Reply:
You are correct that the assumption was that the state does not change in-between.
I have updated the code such that `active` -> `idle` state changes are taken into account (`idle` -> `active` is less a problem since the timeout would be large enough from start).

That works fine, but I think we would need, in a followup,  to think of what to do exactly when there is a state change in the middle for MRP retransmission.

For example, with the following MRP settings:
```
 Idle: 5000ms
 Active: 300ms
```

Assuming the last activity for the node was 2000 ms before the new exchange, we end up with values such as:
```
[1666691506180] [31938:2428400] CHIP: [DIS] Comparaison: 2000 < 4224
[1666691506180] [31938:2428400] CHIP: [DIS] Timeout: (0 uses 300): 412
[1666691506180] [31938:2428400] CHIP: [DIS] Comparaison: 2412 < 4224
[1666691506180] [31938:2428400] CHIP: [DIS] Timeout: (1 uses 300): 412
[1666691506180] [31938:2428400] CHIP: [DIS] Comparaison: 2824 < 4224
[1666691506180] [31938:2428400] CHIP: [DIS] Timeout: (2 uses 300): 659
[1666691506180] [31938:2428400] CHIP: [DIS] Comparaison: 3483 < 4224
[1666691506180] [31938:2428400] CHIP: [DIS] Timeout: (3 uses 300): 1054
[1666691506180] [31938:2428400] CHIP: [DIS] Comparaison: 4537 < 4224
[1666691506180] [31938:2428400] CHIP: [DIS] Timeout: (4 uses 5000): 28147
[1666691506180] [31938:2428400] CHIP: [DIS] Total: 30684
```

While I would have expected something like:
```
[1666691799966] [32228:2431647] CHIP: [DIS] Comparaison: 2000 < 4224
[1666691799966] [32228:2431647] CHIP: [DIS] Timeout: (0 uses 300): 412
[1666691799966] [32228:2431647] CHIP: [DIS] Comparaison: 2412  < 4224
[1666691799966] [32228:2431647] CHIP: [DIS] Timeout: (1 uses 300): 412
[1666691799966] [32228:2431647] CHIP: [DIS] Comparaison: 2824 < 4224
[1666691799966] [32228:2431647] CHIP: [DIS] Timeout: (2 uses 300): 659
[1666691799966] [32228:2431647] CHIP: [DIS] Comparaison: 3483 < 4224
[1666691799966] [32228:2431647] CHIP: [DIS] Timeout: (3 uses 300): 1054
[1666691799966] [32228:2431647] CHIP: [DIS] Comparaison:  4537  < 4224
[1666691799966] [32228:2431647] CHIP: [DIS] Timeout: (4 uses 5000): 6872
[1666691799966] [32228:2431647] CHIP: [DIS] Total: 9409
``

If you look at the last 2 lines of both logs you will noticed that the timeouts differs. The former has a timeout of `28147` while the latter has a timeout of `6872`. It happens because the "last" timeout is considered to be idle and the MRP code uses the number `4` as `sendCount` while this is the first "idle" retransmit.